supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/statuspolling/JesApiQueryManager.scala

```diff
@@ -147,17 +187,25 @@ object JesApiQueryManager {
     def genomicsInterface: Genomics
     def withFailedAttempt: JesApiQuery
     def backoff: Backoff
+    def httpRequest: HttpRequest
+    def contentLength: Long = Option(httpRequest.getContent).map(_.getLength).getOrElse(0L)
   }
   private object JesApiQuery {
     // This must be a def, we want a new one each time (they're mutable! Boo!)
     def backoff: Backoff = SimpleExponentialBackoff(1.second, 1000.seconds, 1.5d)
   }
-  private[statuspolling] final case class JesStatusPollQuery(requester: ActorRef, run: Run, failedAttempts: Int = 0, backoff: Backoff = JesApiQuery.backoff) extends JesApiQuery {
+
+  // Not final so methods can be overridden in unit tests.
```

Comment:
Hmm... You'd want some class that can implement these methods instead of having them directly in the case class, and then in your test, replace the instance of that class with the test version of the methods.

But as I'm thinking about this, it's way more complicated that just allowing the methods to be overridden, so I'd probably leave it as it is, tbh...